src/node.cpp

```diff
@@ -20,19 +20,25 @@ void Node::end_node() {
 }
 
 /* create the logdir if does not exist */
-void create_log_dir() {
+bool create_log_dir() {
   std::string logdir_path = opt_string(LOGDIR);
```

Comment:
```suggestion
  std::string logdir_path(opt_string(LOGDIR));
```

Reply:
With the suggested changes, there was a compilation error. So, I retained the same.